src/node.rs

```diff
@@ -160,14 +169,15 @@ where
                     for i in 0..OCTREE_CHILDREN {
                         if i != octant as usize {
                             let new_octant = Octant::try_from(i).unwrap();
-                            let bounds = self.child_bounds(dimension_3d, new_octant);
-
-                            let mut new_node = Node::<T>::new(bounds);
-                            new_node.ty = NodeType::Leaf(*self.leaf_data().unwrap());
-
+                            let new_node = Node::<T>::new_with_ty(
+                                self.child_bounds(dimension_3d, new_octant),
+                                NodeType::Leaf(T::default()),
+                            );
                             self.children[new_octant as usize] = Box::new(Some(new_node));
                         }
                     }
+                    // move current data to appropriate child ???
+                    let data = std::mem::replace(&mut self.ty, NodeType::Internal);
```

Comment:
Here I have had the understanding that whenever a Leaf node is "promoted" to an internal or simplified Node ( not sure which tbh ), its data will be moved to the child where most appropriate; I just don't understand which child may that be? 

Or the data is supposed to be duplicated at this time?

Reply:
oh I think I got why copy is needed